sql/table.cc

```diff
@@ -8479,6 +8484,199 @@ LEX_CSTRING *fk_option_name(enum_fk_option opt)
   return names + opt;
 }
 
+TR_table::TR_table(THD* _thd, bool rw) : thd(_thd)
+{
+  static const LString table_name("transaction_registry");
+  init_one_table(LEX_STRING_WITH_LEN(MYSQL_SCHEMA_NAME),
+                 XSTRING_WITH_LEN(table_name), table_name, rw ? TL_WRITE : TL_READ);
+  open_tables_backup= new Open_tables_backup;
+  if (open_tables_backup)
+    open_log_table(thd, this, open_tables_backup);
```

Comment:
This function may fail. But as I can read from code it's not a big issue. The only way to handle failures in ctors are exceptions. We can't use them. But maybe it's at least write a warning to log here on failure?

Reply:
`open_log_table()` prints error message. Error state is checked in `update()` and `query()` methods.